.github/workflows/ci.yml

```diff
@@ -352,22 +352,23 @@ jobs:
   merge-check:
     runs-on: ubuntu-latest
     needs:
-      - e2e
-      - bb-native-tests
-      - bb-bench
-      - yarn-project-formatting
-      - yarn-project-test
-      - prover-client-test
+      [
+        e2e,
+        bb-native-tests,
+        bb-bench,
+        yarn-project-formatting,
+        yarn-project-test,
+        prover-client-test,
+        bb-js-test,
+        barretenberg-acir-tests-bb-js,
+        barretenberg-acir-tests-bb,
+        barretenberg-acir-tests-sol,
+      ]
     if: always()
     steps:
       - run: |
-          echo "e2e status: ${{ needs.e2e.result }}"
-          echo "bb-native-tests status: ${{ needs.bb-native-tests.result }}"
-          echo "bb-bench status: ${{ needs.bb-bench.result }}"
-          echo "yarn-project-formatting status: ${{ needs.yarn-project-formatting.result }}"
-          echo "yarn-project-test status: ${{ needs.yarn-project-test.result }}"
-          if [[ "${{ needs.e2e.result }}" != 'success' || "${{ needs.bb-native-tests.result }}" != 'success' || "${{ needs.bb-bench.result }}" != 'success' || "${{ needs.yarn-project-formatting.result }}" != 'success' || "${{ needs.yarn-project-test.result }}" != 'success' ]]; then
-            echo "Pull request merging not allowed due to failures."
+          if [ ${{ contains(needs.*.result, 'failure') }} ]; then
```

Comment:
Nice, didn't know [this was possible](https://docs.github.com/en/actions/learn-github-actions/expressions#example-using-an-object-filter)